gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/PlanServiceImpl.java

```diff
@@ -157,8 +158,12 @@ public PlanEntity create(final ExecutionContext executionContext, final NewPlanE
         try {
             logger.debug("Create a new plan {} for API {}", newPlan.getName(), newPlan.getApiId());
 
-            assertPlanSecurityIsAllowed(executionContext, newPlan.getSecurity().getType());
-            validatePlanSecurity(newPlan.getSecurity().getType(), newPlan.getSecurity().getConfiguration());
+            if (Optional.ofNullable(newPlan.getMode()).orElse(PlanMode.STANDARD) == PlanMode.STANDARD) {
+                assertPlanSecurityIsAllowed(executionContext, newPlan.getSecurity().getType());
+                validatePlanSecurity(newPlan.getSecurity().getType(), newPlan.getSecurity().getConfiguration());
+            } else if (newPlan.getSecurity() != null) {
+                throw new PlanInvalidException("Security type is not required for plan with 'Push' mode");
```

Comment:
"not required" feels like you can provide one security type, I would reverse the negation:
```suggestion
                throw new PlanInvalidException("Security type is not authorized / forbidden for plan with 'Push' mode");
```